torch/_refs/__init__.py

```diff
@@ -534,11 +581,33 @@ def _neg_meta(a: TensorLikeType):
     aten_op=None,  # TODO: this does need a decomp, but kwarg handling is needed
 )
 
+
+def _sigmoid(a: TensorLikeType) -> TensorLikeType:
+    return true_divide(1, add(1, exp(neg(a))))
+
+
+sigmoid = _make_elementwise_unary_reference(
+    _sigmoid,
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.INT_TO_FLOAT,
+    aten_op=torch.ops.aten.sigmoid,
+)
+
 sign = _make_elementwise_unary_reference(
     prims.sign,
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
 )
 
+
+def _signbit(a: TensorLikeType) -> TensorLikeType:
+    return lt(a, 0)
```

Comment:
I was worried this would be incorrect because of -0 but the situation is more complicated

```
a = torch.tensor((-0.))
torch.lt(a, 0)
: tensor(False)

torch.signbit(a)
: tensor(False)

a = np.array((-0.))
np.signbit(a)
: True

a = np.array((-0.))
a < 0
: False

a = np.array((-0.))
jnp.signbit(a)
: DeviceArray(True, dtype=bool)

a = np.array((-0.))
jax.lax.lt(a, 0.)
: DeviceArray(False, dtype=bool)

```

So our signbit function is implemented incorrectly! We actually have an issue for this, too: https://github.com/pytorch/pytorch/issues/53963. cc @kshitij12345. It'd be nice to fix eager signbit (but it doesn't have to be in this PR), but we should cite this issue. The decomposition can be left as-is for now as long as we update the issue stating the decomposition is consistent but incorrect, and when fixing eager signbit we should also fix this. A more accurate decomposition would probably acquire the signbit directly through bit manipulation or create a separate mask for special values like -0 (are there others we need to consider)? 

In addition to that:
- let's have this throw a custom error for complex dtypes (they can't be compared with zero)
- and let's short-circuit to just return full_like(a, False) for when a is a boolean tensor or an unsigned type

Reply:
I changed eager mode to use the `signbit` c++/cuda function and made `signbit` a prim.

Reply:
Nice!